Updated React Router Codeblock to v6 (#454)
Codecademy/docs: content/react/concepts/routing/routing.md

- `react-router-native`: React Native version
- `react-router-dom`: Recommended for web applications

Install the newest version with `npm`:

```

> npm install react-router-dom@latest

Reviewer: Love this, it's more future-proof 👍

```

> The newest version of React Router is v6, which comes with many breaking changes including:

Reviewer 2: What do we think about linking out to the official overview page for React Router v6? Like so: 

The newest version of React Router is [v6](https://reactrouter.com/docs/en/v6/getting-started/overview), which comes with many breaking changes including:

Reviewer: Ditto on this point

> The newest version of React Router is v6, which comes with many breaking changes including:

Reviewer: ```suggestion
The newest version of React Router is [v6](https://reactrouter.com/docs/en/v6/getting-started/overview), which comes with many breaking changes including:
```

- Using a `<Routes />` component instead of a `<Switch />` component for `<Route />` configuration.
- Using the `element` attribute instead of `component` when passing a JSX to a `<Link />`.

## Example

> Below, a `<BrowserRouter/>` (aliased as `<Router/>`) is used to map the navigation `<Link/>` components to a `<Routes/>` component. The `<Routes/>` renders the appropriate `<Route>` based on the most specific-matching `path` among all possible matches .

Reviewer: We can, however, include the distinct changes like how the order that the route paths are listed no longer takes precedence over the most specific-matching path.

```suggestion
Below, a `<BrowserRouter/>` (aliased as `<Router/>`) is used to map the navigation `<Link/>` components to a `<Routes/>` component. The `<Routes/>` renders the appropriate `<Route>` based on the most specific-matching `path` among all possible matches.
```

Extra space before the period. If you want to include my point mentioned above, it can be added at the end of the sentence or as a **Note**: In older versions, the order that the route paths are listed took precedence over the most specific-matching path.

This note can still be helpful for people transitioning to newer version

> 

Reviewer 2: ```suggestion

The newest version of React Router is v6, which comes with many breaking changes including: 

- Using a `<Routes />` component instead of a `<Switch />` component for `<Route />` configuration.
- Using the `element` attribute instead of `component` when passing a JSX to a `<Link />`.
```

> Below, a `BrowserRouter` is used to map the navigation `<Link/>` components to a `<Switch/>` component. The `<Switch/>` renders the appropriate `<Route>` based on a top-down search and match between the `to` attribute in the `Link` and the `path` attribute in the `Route`.

Reviewer 2: ```suggestion
Below, a `<BrowserRouter/>` (aliased as `<Router/>`) is used to map the navigation `<Link/>` components to a `<Routes/>` component. The `<Routes/>` renders the appropriate `<Route>` based on the most specific-matching `path` among all possible matches .
```

> ```jsx

Reviewer 2: I think that everything from the beginning of this file up to this line needs a second look since the text above this code snippet. It should be edited to include only language having to do with the latest version of React Router, v6. (for example, remove any mention of '<Switch/>`)

import React from 'react';

> import { Routes, Route, Link } from 'react-router-dom';

Reviewer 2: ```suggestion
import {
  BrowserRouter as Router,
  Routes,
  Route,
  Link
} from 'react-router-dom';
```

According to the [new docs on React Router](https://reactrouter.com/docs/en/v6/getting-started/overview) we still need to use `<BrowserRouter>` so that `<Routes>` is being used in the correct context. Without it, I'm getting this error:

```
Error: useRoutes() may be used only in the context of a <Router> component.
```

const App = () => {
  return (
    <div>

>       <nav>
>         <ul>
>           <Routes>
>             <Route path="/about" element={<About />} />
>             <Route path="/projects" element={<Projects />} />
>             <Route path="/contact" element={<Contact />} />
>             <Route path="/" element={<Home />} />
>           </Routes>
>           <li>
>             <Link to="/">Home</Link>
>           </li>
>           <li>
>             <Link to="/about">About</Link>
>           </li>
>           <li>
>             <Link to="/projects">Projects</Link>
>           </li>
>           <li>
>             <Link to="/contact">Contact</Link>
>           </li>
>         </ul>
>       </nav>

Reviewer 2: ```suggestion
      <Router>
        <nav>
          <ul>
            <Routes>
              <Route path="/about" element={<About />} />
              <Route path="/blog" element={<Blog />} />
              <Route path="/contact" element={<Contact />} />
              <Route path="/" element={<Home />} />
            </Routes>
            <li>
              <Link to="/">Home</Link>
            </li>
            <li>
              <Link to="/about">About</Link>
            </li>
            <li>
              <Link to="/projects">Projects</Link>
            </li>
            <li>
              <Link to="/contact">Contact</Link>
            </li>
          </ul>
        </nav>
      </Router>
```

Using the `<Router> (<BrowserRouter>)` component does away with the error I'd mentioned before.
